test/reco_particle_ref.cc

```diff
@@ -137,7 +136,7 @@ void writeRecoParticleRef(const TestState& testState) {
     int i = 0;
     for (const auto& ref : refColl) {
       auto expectedParticle = recoColl[testState.relationIndices[iEvent][i]];
-      ASSERT_EQUAL(ref.getParticle(), expectedParticle, "Wrong particle reference during creation");
+      ASSERT_EQUAL(expectedParticle, expectedParticle, "Wrong particle reference during creation");
```

Comment:
```suggestion
      ASSERT_EQUAL(ref, expectedParticle, "Wrong particle reference during creation");
```
This should obviously be comparing two different things, not the same thing with itself.